juniper_official/interface/check-ethernet-crc-framing-errors.rule

```diff
@@ -0,0 +1,218 @@
+ * Two inputs control detection
+ *
+ *   1) interface-name, matches the interfaces
+ *      that you would like to monitor.  By default it's '.*', which matches
+ *      all interfaces. Use something like 'ge.*' to match only gigabit
+ *      ethernet interfaces. 
+ *
+ *   2) discard-count, usef to check if the error count is increasing.
+ *      By default it's 1.
+ * 
+ * 
+ */
+healthbot {
+    topic interfaces {
+        rule check-ethernet-crc-framing-errors {
+            /*
+             * Monitors CRC,In-Fragment and Jabber frame errors and notifies via dashboard, red color 
+             * when threshold is exceeded else green
+			 *
+             * Use interface-name as key for rule.
+             */
+            keys interface-name;
+            synopsis "Ethernet CRC and Frames Error KPI's";
+            description "Collects CRC and Frames errors and notifies when count increases";
+            /*
+             * Sensor configuration to get data from network devices.
+             */			
+            sensor interface-sensor-oc {
+                synopsis "open-config sensor definition";
+                description "Collects data from network device by subscribing to oc sensor paths";
+                open-config {
+                    sensor-name /interfaces/interface/ethernet;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined using sensor path. Map the longer sensor names
+             * to the shorter field names used in the rules.
+             */				
+            field crc-errors {
+                sensor interface-sensor-oc {
+                    path /interfaces/interface/ethernet/state/counters/in-crc-errors;
+					zero-suppression;
+                }
+                type integer;
+                description "Cyclic Redundancy Check errors";
+            }
+            field drop-threshold {
+                constant {
+                    value "{{discard-count}}";
+                }
+                type integer;
+                description "Drop count increase threshold";
+            }
+            field fragment-frames {
+                sensor interface-sensor-oc {
+                    path /interfaces/interface/ethernet/state/counters/in-fragment-frames;
+					zero-suppression;
+                }
+                type integer;
+                description "fragement frames";
+            }
+            field interface-name {
+                sensor interface-sensor-oc {
+                    where "/interfaces/interface/@name =~ /{{interface-name}}/";
+                    path "/interfaces/interface/@name";
+                }
+                type string;
+                description "Interfaces to be monitored";
+            }
+            field jabber-frames {
+                sensor interface-sensor-oc {
+                    path /interfaces/interface/ethernet/state/counters/in-jabber-frames;
```

Comment:
Can you check if this is required. Get confirmation from PLM if required

Reply:
Replacing this with different rule to collect for fec error along with crc and framing errors.